src/sage/sets/disjoint_set.pyx

```diff
@@ -779,8 +826,44 @@ cdef class DisjointSet_of_hashables(DisjointSet_class):
             ...
             KeyError: 5
         """
-        i = self._el_to_int[e]
-        r = self._d.find(i)
+        cdef int i = self._el_to_int[e]
+        cdef int r = self._d.find(i)
+        return self._int_to_el[r]
```

Comment:
I don't think there is a benefit to typecasting here. It might actually be slower since it has to check the types.